assets/js/components/ModulesList.js

```diff
@@ -0,0 +1,180 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import map from 'lodash/map';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	showErrorNotification,
+	trackEvent,
+} from '../util';
+import Data from 'googlesitekit-data';
+import Link from './Link';
+import ModuleIcon from './ModuleIcon';
+import GenericError from './notifications/generic-error';
+import ModuleSettingsWarning from './notifications/module-settings-warning';
+import VisuallyHidden from './VisuallyHidden';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ModulesList( { moduleSlugs } ) {
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const modulesData = useSelect( ( select ) => select( CORE_MODULES ).getModules() ?? {} );
+
+	/**
+	 * Handle setup module click event.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param {string} slug Module slug.
+	 */
+	const handleSetupModule = useCallback( async ( slug ) => {
+		try {
+			const { response } = await activateModule( slug );
+			await trackEvent(
+				`${ slug }_setup`,
+				'module_activate',
+				slug,
+			);
+
+			// Redirect to ReAuthentication URL
+			global.location.assign( response.moduleReauthURL );
+		} catch ( err ) {
+			showErrorNotification( GenericError, {
+				id: 'setup-module-error',
+				title: __( 'Internal Server Error', 'google-site-kit' ),
+				description: err.message,
+				format: 'small',
+				type: 'win-error',
+			} );
+		}
+	}, [ activateModule ] );
+
+	// Filter specific modules
+	const moduleObjects = Array.isArray( moduleSlugs ) && moduleSlugs.length
+		? moduleSlugs
+			.filter( ( slug ) => modulesData.hasOwnProperty( slug ) )
+			.reduce( ( acc, slug ) => ( { ...acc, [ slug ]: modulesData[ slug ] } ), {} )
+		: modulesData;
+
+	// Filter out internal modules.
+	const modules = Object.values( moduleObjects ).filter( ( module ) => ! module.internal );
+
+	// Map of slug => name for every module that is active and completely set up.
+	const completedModuleNames = modules
+		.filter( ( module ) => module.active && module.connected )
+		.reduce( ( completed, module ) => {
+			completed[ module.slug ] = module.name;
+			return completed;
+		}, {} );
+
+	// Sort modules and exclude those that required other modules to be set up.
+	// Logic still in place below in case we want to add blocked modules back.
+	const sortedModules = modules
+		.filter( ( module ) => 0 === module.dependencies.length )
+		.sort( ( module1, module2 ) => module1.sort - module2.sort );
+
+	return (
+		<div className="googlesitekit-modules-list">
+			{ map( sortedModules, ( module ) => {
+				let blockedByParentModule = false;
+				let parentBlockerName = '';
+				const {
+					slug,
+					name,
+					connected,
```

Comment:
@kostyalmm  sorry for the incorrect suggestion at the beginning, but here we need to rely on the `active` property instead of `connected`.

Reply:
Are you sure? 🤔 `connected` seems to be the correct property here according to me as the [old component has](https://github.com/google/site-kit-wp/blob/develop/assets/js/components/modules-list.js#L119):
```
setupComplete: isConnected,
```

By this logic, `connected` seems like the right property. What do you think?

Reply:
@kostyalmm as far as I understand it, `connected` means that the module is activated and the user has configured which is not what we need here. We need to know whether a module is activated and if it's not, let the user activate it by clicking on the "Connect" link.

@aaemnnosttv am I right?

Reply:
@eugene-manuilov `setupComplete` for the legacy module format is equal to `module.active && module.connected`:
https://github.com/google/site-kit-wp/blob/27d4df7bbfbf6c7e59310ee9338e2a6155870268/includes/Core/Modules/Modules.php#L159

So in this case, it's not a 1-to-1 equivalent logically but it is the closest thing for `setupComplete`.

With that said, the consuming logic would _also_ need to check for `active` as well whereas before it was included in the definition of `setupComplete`.

cc @kostyalmm

Reply:
Thanks, @aaemnnosttv! @kostyalmm let's add `active` and use `( active && connected )` to determine whether a module is connected or not.